internal/controller/observabilityalertrule/controller.go

```diff
@@ -280,6 +293,61 @@ func setStatusCondition(rule *openchoreov1alpha1.ObservabilityAlertRule, status
 	rule.Status.Conditions = append(rule.Status.Conditions, newCond)
 }
 
+// finalize handles the cleanup of the alert rule from the observer backend
+func (r *Reconciler) finalize(ctx context.Context, alertRule *openchoreov1alpha1.ObservabilityAlertRule) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+
+	if !controllerutil.ContainsFinalizer(alertRule, AlertRuleCleanupFinalizer) {
+		// Finalizer not present, nothing to clean up
+		return ctrl.Result{}, nil
+	}
+
+	logger.Info("Deleting alert rule from observer backend", "name", alertRule.Name, "sourceType", alertRule.Spec.Source.Type)
+
+	// Call DELETE endpoint on observer
+	observerURL := getObserverBaseURL()
+	url := fmt.Sprintf("%s%s/%s/%s", observerURL, alertingRulePath, alertRule.Spec.Source.Type, alertRule.Name)
+
+	reqCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
```

Comment:
We should consider extracting these calls into a observer client in a follow-up PR.